crates/gitbutler-settings/src/lib.rs

```diff
@@ -9,3 +15,42 @@ pub struct AppSettings {
     /// Client ID for the GitHub OAuth application
     pub github_oauth_client_id: Option<String>,
 }
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default, PartialEq, Hash)]
+pub struct AppSettingsUpdateRequest {
```

Comment:
Is it possible to apply `#[serde(default, skip_serializing_if = "Option::is_none")]` to the struct itself?

Reply:
I am not sure I am following here

Reply:
I have never seen it on the struct itself, but if there was a `field: AppSettingsUpdateRequest` it would certainly be possible to implement something like it.
Right now I see it used only on the top-level of `tauri` function calls.
Probably I am missing something though.

Reply:
I don't understand the issue tbh - having the update request contain Option fields allows us to do partial updating without having to pass the entire settings struct over, which I think is desirable. @mtsgrd can you clarify what you meant?

Reply:
What I mean is, if you have a struct with a bunch of optional values then on serializing that it will write out all keys, and null values for them. Instead, it would be great if it omits unset fields.

Reply:
But this is the request struct that the UI/Javascript will create and the rust end will consume in order to perform updates. It looks to me there is no ambiguity on either end, unless I am missing something still

Reply:
Oh shoot, you're right. I think I mean to comment on the `AppSettings` struct.

Reply:
Oh I see. But even still, I imagine we want to be able to express and know that a config exists but has not been set? I thought we can do something like this on the frontend
```ts
export class AppSettings {
  appMetricsEnabled: boolean | undefined
}
```
Why is not serializing `app_metrics_enabled` here helpful? We have other places in the code where `Option<bool>` in the rust struct becomes `boolean | undefined` on the javascript end - is there a flaw with that that I don't know about?

Reply:
Lol, sorry, I could be more clear from start. I just don't want the `settings.json` file to contain every possible key unless set. Wdyt?

Reply:
Oh yeah, that makes total sense. We anyways need to make sure that the settings is backwards / forwards compatible. 
But - I sense the reason you are saying this is for user edit-ability. But this setup is inherently not appropriate for that. I wanna make this friendly for a user to modify and screw up, but not sure yet how to build it in a nice way. This was actually the root discussion point i had here https://github.com/gitbutlerapp/gitbutler/pull/5774#issuecomment-2525762673

Reply:
Editability yes, but also readability and to make settings easier to share. Most of the time I edit vscode settings in the UI, but every once in a while I copy paste something in that I found online or got in a chat.

While we're on this topic. Could there be any settings we want to share at the repo level? Similar to how `.vscode` is used.

Reply:
I guess what I am saying is - I would like to make a separation between the settings and the file that a user reads and updates. Have them be two separate things, such that it's both safer to update and as you said easier to read. 
I am not sure yet what is the best way to implement that, but this PR was not trying to implement this, but instead just move us away from that tauri store plugin.

Regarding project settings - yes it will be great to move away from projects.json but let's nail app settings first, then it will be easy to extend from there

Reply:
Just to clarify and summarize - the `settings.json` in it's current form is _not_ appropriate for user editing. 
The reason for that is not just the readability and serializing the None values, but also the fact that it is fragile - if this file doesn't parse, the app will not start.

I would love to move to a `settings.json` VSCode style, and not serializing unset values is necessary but not sufficient to getting there. I was initially thinking of this PR as simple change to get off the tauri store plugin, but since you fixed the state issue in the frontend, perhaps the scope of this can be expanded to land us in the endgame setup